tests/emulator/src/vbridge_impl.cc

```diff
@@ -128,12 +134,20 @@ std::optional<SpikeEvent> VBridgeImpl::create_spike_event(insn_fetch_t fetch) {
   // create SpikeEvent
   uint32_t opcode = clip(fetch.insn.bits(), 0, 6);
   uint32_t width = clip(fetch.insn.bits(), 12, 14);
+  uint32_t rs1 = clip(fetch.insn.bits(), 15, 19);
+  uint32_t csr = clip(fetch.insn.bits(), 20, 31);
+
   // for load/store instr, the opcode is shared with fp load/store. They can be only distinguished by func3 (i.e. width)
   // the func3 values for vector load/store are 000, 101, 110, 111, we can filter them out by ((width - 1) & 0b100)
   bool is_load_type  = opcode == 0b0000111 && ((width - 1) & 0b100);
   bool is_store_type = opcode == 0b0100111 && ((width - 1) & 0b100);
   bool v_type = opcode == 0b1010111 && width != 0b111;
-  if (is_load_type || is_store_type || v_type) {
+
+  bool csr_type = opcode == 0b1110011 && (width & 0b011);
+  bool csr_write = csr_type && ((width & 0b100) | rs1);
```

Comment:
I prefer using `is_xxx` variable name for boolean

Reply:
Ack